sdk/src/record-provider.ts

```diff
@@ -101,7 +156,7 @@ interface RecordProvider {
      *
      * const record = await recordProvider.findRecord(true, [], params);
      */
-    findRecord(unspent: boolean, nonces?: string[], searchParameters?: RecordSearchParams): Promise<RecordPlaintext>;
+    findRecord(searchParameters: RecordSearchParams, filterFn?: (record: RecordPlaintext) => boolean, nonces?: string[]): Promise<OwnedRecord>;
```

Comment:
```suggestion
    findRecord(searchParameters: RecordSearchParams): Promise<OwnedRecord>;
```

While I understand what a predicate might do, I think it complicates the method signature and the interface too much, so I'm going to request it's removal. Same for the below.

Reply:
done